### PR TITLE
removed junit dependency and included assertj for assertion

### PR DESCRIPTION
**Pull request by ajoecker**
see https://github.com/getgauge/gauge-java/issues/155

assertion from junit is replaced with assertj (http://joel-costigliola.github.io/assertj/index.html)

### CODE REVIEW

**Comment by sriv** [comment]
@ajoecker - thanks for picking this up. Looks like there are some conflicts, could you please have a look?

**Comment by ajoecker** [comment]
sure :)

File: gauge-archetype-selenium/src/test/resources/projects/test1/reference/src/test/java/StepImplementation.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        assertThat(instructions).isNotNull();
+    }
+
+    @Step("Open the Gauge homepage")
+    public void implementation1() {
+        String app_url = System.getenv("APP_URL");
+        Driver.webDriver.get(app_url + "/");
+        assertThat(Driver.webDriver.getTitle()).contains("Gauge");
+    }
+}
```
Comment (Apoorva-GA) [inline gauge-archetype-selenium/src/test/resources/projects/test1/reference/src/test/java/StepImplementation.java]: This file is not needed. Could you remove this.

File: gauge-archetype-selenium/src/main/resources/archetype-resources/src/test/java/StepImplementation.java [hunk 2]
```diff
@@ -6,27 +6,27 @@
...
-  @Step("Ensure installation instructions are available")
-  public void ensureInstallationInstructionsAreAvailable() throws InterruptedException {
-      WebElement instructions = Driver.webDriver.findElement(By.xpath("//p[@class='instruction']"));
-      assertTrue(instructions!=null);
-  }
+    @Step("Ensure installation instructions are available")
+    public void ensureInstallationInstructionsAreAvailable() throws InterruptedException {
+        WebElement instructions = Driver.webDriver.findElement(By.xpath("//p[@class='instruction']"));
+        assertThat(instructions).isNotNull();
+    }
```
Comment (Apoorva-GA) [inline gauge-archetype-selenium/src/main/resources/archetype-resources/src/test/java/StepImplementation.java]: Could you format this file. There is a new space added before each line.

File: gauge-archetype-selenium/src/test/resources/projects/test1/reference/src/test/java/com/test/pkg/StepImplementation.java [hunk 3]
```diff
@@ -6,27 +6,27 @@
...
-      String app_url = System.getenv("APP_URL");
-      Driver.webDriver.get(app_url + "/");
-      assertTrue(Driver.webDriver.getTitle().contains("Gauge"));
-  }
+    @Step("Open the Gauge homepage")
+    public void implementation1() {
+        String app_url = System.getenv("APP_URL");
+        Driver.webDriver.get(app_url + "/");
+        assertThat(Driver.webDriver.getTitle()).contains("Gauge");
+    }
```
Comment (Apoorva-GA) [inline gauge-archetype-selenium/src/test/resources/projects/test1/reference/src/test/java/com/test/pkg/StepImplementation.java]: Could you format this file. There is a new space added before each line.

**Pull request merged**